src/pages/ws/ui/variant-drawer/variant-drawer-header/drawer-layout-control/drawer-layout-control.tsx

```diff
@@ -1,48 +1,92 @@
 import styles from './drawer-layout-control.module.css'
 
-import { ReactElement, useState } from 'react'
+import { ReactElement, useEffect, useState } from 'react'
 import cn from 'classnames'
 
 import { useModal } from '@core/hooks/use-modal'
+import { usePopover } from '@core/hooks/use-popover'
 import { t } from '@i18n'
 import { Button } from '@ui/button'
 import { Divider } from '@ui/divider'
 import { Icon } from '@ui/icon'
+import { SolutionChangeDialog } from '@components/solution-control/solution-change-dialog'
+import { showToast } from '@utils/notifications'
 import {
   IVariantDrawerGridPreset,
   VariantDrawerLayoutMode,
 } from '../../variant-drawer.interface'
-import { PresetMenu } from './preset-menu'
+import { PresetMenuPopover } from './preset-menu'
 import { SavePresetDialog } from './save-preset-dialog'
 
 interface IDrawerLayoutControlProps {
-  className?: string
   layoutMode: VariantDrawerLayoutMode
-  onChangeLayoutMode: (mode: VariantDrawerLayoutMode) => void
   gridPresets: IVariantDrawerGridPreset[]
+  windowsOpenState: boolean
+  appliedPreset: string | null
+  className?: string
+  onChangeLayoutMode: (mode: VariantDrawerLayoutMode) => void
   onSaveGridPreset?: (presetName: string) => void
   onChangeGridPreset: (presetName: string) => void
-  windowsOpenState: boolean
+  onModifyGridPreset: (presetName: string) => void
+  onDeleteGridPreset: (presetName: string) => void
   onWindowsToggle: (state: boolean) => void
 }
 
 export const DrawerLayoutControl = ({
   className,
   gridPresets,
   layoutMode,
+  appliedPreset,
   onChangeLayoutMode,
   onSaveGridPreset,
   onChangeGridPreset,
+  onModifyGridPreset,
+  onDeleteGridPreset,
   windowsOpenState,
   onWindowsToggle,
 }: IDrawerLayoutControlProps): ReactElement => {
-  const [presetMenuAnchorEl, setPresetMenuAnchorEl] =
-    useState<HTMLElement | null>(null)
   const [savePresetDialog, openSavePresetDialog, closeSavePresetDialog] =
     useModal()
 
-  const closePresetMenu = () => {
-    setPresetMenuAnchorEl(null)
+  const [deleteDialog, openDeleteDialog, closeDeleteDialog] = useModal()
+
+  const { popoverAnchor, isPopoverOpen, onToggle, closePopover } = usePopover()
+
+  const [selectedPreset, setSelectedPreset] = useState<string>('')
+
+  useEffect(() => {
+    if (isPopoverOpen && selectedPreset !== appliedPreset) {
+      setSelectedPreset(appliedPreset || '')
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Why you didn't add other dependencies?

Reply:
Cause they will clear the selected preset on every select